packages/cli/src/models/compiler/solidity/CompilerProvider.ts

```diff
@@ -0,0 +1,212 @@
+import fs from 'fs';
+import { mkdirpSync, readJsonSync, writeJsonSync } from 'fs-extra';
```

Comment:
Well, since we are doing everything on a single execution green thread, we would end up awaiting on it anyway without doing anything else. If there were other operations to parallelize, I'd agree, but I see no need for awaiting on js here.

Reply:
Slack thread 😄